web/components/demo/index.js

```diff
@@ -19,6 +19,17 @@ const FrontEndComponents = () => (
       &nbsp;
       <SecondaryButton text="Secondary" onClick={ButtonCallback} disabled />
     </div>
+    <br />
+    <div>
+      <ProgressGroup
+        activeIndex={1}
+        steps={[
+          { text: '1', onClick: ButtonCallback },
+          { text: '2', onClick: ButtonCallback },
+          { text: '3', onClick: ButtonCallback, disabled: true },
```

Comment:
Can we move the disable logic to the progress group component? Ideally we would only pass in activeIndex and number of buttons. And just disable all buttons with index larger than activeIndex

Reply:
I was thinking about this too, but `activeIndex` could be less than the most activate-able index. For example, you could get to step 3, then go to step 2, but you want step 3 to still be clickable.

What do you think about:

```js
ProgressGroup.propTypes = {
  activeIndex: PropTypes.number, // "clicked"
  lastActivatableIndex: PropTypes.number, // all values after this will be "disabled"
  steps: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func,
  })),
};
```

Reply:
I don’t think steps is necessary - you can automatically generate n number of buttons labeled from 1 to n, given that you pass in n in the props. And we’ll need a callback function that is called whenever the page is changed so that whoever is listening to that callback can know what page it’s in.

And whether we can click a page that’s after the current page is something we should confirm with design

Reply:
I'll go with my second idea then, and just allow one callback that all the button clicks call which provides the current page index.